src/app/correlation-id/correlation-id.service.ts

```diff
@@ -24,15 +36,32 @@ export class CorrelationIdService {
     protected cookieService: CookieService,
     protected uuidService: UUIDService,
     protected store: Store<AppState>,
+    protected orejimeService: OrejimeService,
+    @Inject(NativeWindowService) protected _window: NativeWindowRef,
   ) {
+    if (this._window?.nativeWindow) {
+      this._window.nativeWindow.initCorrelationId = () => this.initCorrelationId();
+    }
   }
 
   /**
-   * Initialize the correlation id based on the cookie or the ngrx store
+   * Check if the correlation id is allowed to be set, then set it
    */
   initCorrelationId(): void {
+    this.orejimeService?.getSavedPreferences().subscribe(preferences => {
+      if (preferences[CORRELATION_ID_OREJIME_KEY]) {
```

Comment:
If preferences is undefined, you get an error in the console:

```suggestion
      if (hasValue(preferences) && preferences[CORRELATION_ID_OREJIME_KEY]) {
```